src/VisualStudio/Core/Def/Implementation/Preview/PreviewUpdater.cs

```diff
@@ -41,24 +40,26 @@ public void CloseWorkspace()
 
         public void UpdateView(TextDocument document, SpanChange spanSource)
         {
-            var documentText = document.GetTextAsync().Result.ToString();
-            if (TextView.TextBuffer.CurrentSnapshot.GetText() != documentText)
-            {
-                UpdateBuffer(document, spanSource);
-            }
+            UpdateBuffer(document);
 
             // Picking a different span: no text change; update span anyway.
-            SpanToShow = spanSource.GetSpan();
-            var spanInBuffer = new SnapshotSpan(TextView.TextBuffer.CurrentSnapshot, new Span(SpanToShow.Start, 0));
-            TextView.ViewScroller.EnsureSpanVisible(spanInBuffer, EnsureSpanVisibleOptions.None);
-            Tagger.OnTextBufferChanged();
+            _tagger.Span = spanSource.GetSpan();
```

Comment:
What happens when the TagsChanged event is fired? Does it matter that this now happens before the scroll?

Reply:
When we fire tags changed, it means the editor will query back for us where they should be doing the current highlighting. I guess the ordering doesn't actually matter here, since the editor all will query if scrolling happens or something else that requires a relayout of text. I guess this probably still slightly better as otherwise we might be triggering an extra layout when we raise tags changed when we haven't scrolled to where the user wants to see, so it'd just be wasting time.